src/main/java/org/sagebionetworks/web/client/OneSageUtilsImpl.java

```diff
@@ -9,28 +9,54 @@
 public class OneSageUtilsImpl implements OneSageUtils {
 
   private final GWTWrapper gwtWrapper;
+  private final SynapseProperties synapseProperties;
 
   @Inject
-  public OneSageUtilsImpl(GWTWrapper gwtWrapper) {
+  public OneSageUtilsImpl(
+    GWTWrapper gwtWrapper,
+    SynapseProperties synapseProperties
+  ) {
     this.gwtWrapper = gwtWrapper;
+    this.synapseProperties = synapseProperties;
   }
 
   private String getHostForOneSage() {
     // SWC-6533: We do not want to stack hop for Prod and Staging
+
+    if (synapseProperties.getIsDevMode()) {
+      // If in dev mode, redirect to port 3000 (might be remote host)
+      return (
+        Window.Location.getProtocol() +
```

Comment:
Since this string is now formulated twice, this could be refactored into a shared (private) method